src/main/java/com/autotune/common/data/datasource/DataSourceFactory.java

```diff
@@ -65,14 +65,15 @@ public static DataSource getDataSource(String dataSource) throws MonitoringAgent
 	 */
 	private static String getMonitoringAgentEndpoint() throws MonitoringAgentNotFoundException {
 		//No endpoint was provided in the configmap, find the endpoint from the service.
-		KubernetesClient client = new DefaultKubernetesClient();
-		ServiceList serviceList = client.services().inAnyNamespace().list();
+		KubernetesServices kubernetesServices = new KubernetesServicesImpl();
```

Comment:
I think this should be done in the `AutotuneDeploymentInfo` based on the type of the kubernetes cluster that we are deployed in. Eg if it is OpenShift/minikube, then accordingly initiate the right impl

Reply:
Yes. I will raise separate PR and get it implemented using Metrics/query abstraction.